src/ape/contracts/base.py

```diff
@@ -677,6 +677,74 @@ def _mutable_methods_(self) -> Dict[str, ContractTransactionHandler]:
             # NOTE: Must raise AttributeError for __attr__ method or will seg fault
             raise AttributeError(str(err)) from err
 
+    def call_view_method(self, method_name: str, *args, **kwargs) -> Any:
+        """
+        Call a contract's view function directly using the method_name.
+        This is helpful in the scenario where the contract has a
+        method name matching an attribute of the BaseAddress class,
+        such as 'nonce' or 'balance'
+
+        Args:
+            method_name (str): The contract method name to be called
+            *args: Contract method arguments.
+            **kwargs: Transaction values, such as ``value`` or ``sender``
+
+        Returns:
+            Any: Output of smart contract view call.
+
+        """
+
+        if method_name in self._view_methods_:
+            view_handler = self._view_methods_[method_name]
+            output = view_handler(*args, **kwargs)
+            return output
+
+        elif method_name in self._mutable_methods_:
+            mutable_handler = self._mutable_methods_[method_name]
```

Comment:
I think we have to do something like this:

```suggestion
            handler = self._mutable_methods_[method_name].call
```

in order to handle mutable methods as calls.

Let's add tests to be sure.